File: Javascript/problems/problem002.js
function problem002(target) {
    let sum = 0;
    let a = 1;
    let b = 1;
    let c = a + b;

    while(c < target) {
        sum += c;
        a = b + c;
        b = c + a;
        c = a + b;
    }
    
    return sum;
}

module.exports = problem002;

File: C#/src/ProjectEulerSolutions/EulerProblems/Problem002.cs
namespace ProjectEulerSolutions.EulerProblems
{
    public class Problem002
    {
        public static int Solution(int len)
        {
            int sum = 2;
            int num1 = 1;
            int num2 = 2;
            int fibonacci = num1 + num2;
            
            while (fibonacci < len)
            {
                num1 = num2;
                num2 = fibonacci;

                fibonacci = num1 + num2;

                if (fibonacci % 2 == 0)
                {
                    sum += fibonacci;
                }
            }

            return sum;
        }
    }
}
